phpunit/class-gutenberg-rest-templates-controller-test.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Unit tests covering Gutenberg_REST_Templates_Controller_6_7 functionality.
+ */
+class Gutenberg_REST_Templates_Controller_6_7_Test extends WP_Test_REST_Controller_Testcase {
+
+	/**
+	 * @var int
+	 */
+	protected static $admin_id;
+
+	public static function wpSetupBeforeClass( $factory ) {
+		self::$admin_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	public static function wpTearDownAfterClass() {
+		self::delete_user( self::$admin_id );
+	}
+
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+
+		$template_name = 'test-plugin//test-template';
+		$args          = array(
+			'content'     => 'Template content',
+			'title'       => 'Test Template',
+			'description' => 'Description of test template',
+			'post_types'  => array( 'post', 'page' ),
+		);
+
+		wp_register_template( $template_name, $args );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/templates/test-plugin//test-template' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertNotWPError( $response );
+
+		$data = $response->get_data();
+
+		$this->assertEquals( 'default//test-template', $data['id'] );
+		$this->assertEquals( 'default', $data['theme'] );
+		$this->assertEquals( 'Template content', $data['content']['raw'] );
+		$this->assertEquals( 'test-template', $data['slug'] );
+		$this->assertEquals( 'plugin', $data['source'] );
+		$this->assertEquals( 'plugin', $data['origin'] );
+		$this->assertEquals( 'Description of test template', $data['description'] );
+		$this->assertEquals( 'Test Template', $data['title']['rendered'] );
+		$this->assertEquals( 'test-plugin', $data['plugin'] );
```

Comment:
These ought to be `assertSame` and will need messages to assist contributors to determine which test has failed.

Reply:
Good points, thanks! In 069672cd0b2b3d73785684603cc65d277b73301c I switched from `assertEquals()` to `assertSame()` when possible and added error messages when there was more than one assertion in the same test.